app/src/main/java/org/glucosio/android/activity/FreestyleLibre.java

```diff
@@ -0,0 +1,281 @@
+        /**
+         * Call this before onPause, otherwise an IllegalArgumentException is thrown as well.
+         */
+        stopForegroundDispatch(this, mNfcAdapter);
+
+        super.onPause();
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        /**
+         * This method gets called, when a new Intent gets associated with the current activity instance.
+         * Instead of creating a new activity, onNewIntent will be called. For more information have a look
+         * at the documentation.
+         *
+         * In our case this method gets called, when the user attaches a Tag to the device.
+         */
+        handleIntent(intent);
+    }
+
+    private void handleIntent(Intent intent) {
+        String action = intent.getAction();
+        if (NfcAdapter.ACTION_TECH_DISCOVERED.equals(action)) {
+
+            Log.d("glucosio", "NfcAdapter.ACTION_TECH_DISCOVERED");
+            // In case we would still use the Tech Discovered Intent
+            Tag tag = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
+            String[] techList = tag.getTechList();
+            String searchedTech = NfcV.class.getName();
+            new NfcVReaderTask().execute(tag);
+
+        }
+    }
+
+    private void showReadingLayout() {
+/*        SimpleDatabase database = new SimpleDatabase(this);
+        long id = database.saveMessage(mResult);
+        ReadingData.TransferObject transferObject = new ReadingData.TransferObject(id, mResult);
+        database.close();
+        WearableApi.sendMessage(mGoogleApiClient, WearableApi.GLUCOSE, new Gson().toJson(transferObject), mMessageListener);
+        mMessagesBeingSent++;
+        mFinishAfterSentMessages = true;*/
+
+        // Apply values in TextViews
+        // TODO: Add check for mmol/L
+        readingTextView.setText(mResult.history.get(mResult.history.size()-1).glucose(false));
+
+        new Runnable() {
+            @Override
+            public void run() {
+                View view = findViewById(R.id.activity_freestyle_reading);
+                view.setVisibility(View.INVISIBLE);
+                AnimationTools.startCircularReveal(view);
+            }
+        }.run();
+        // Reveal ReadingLayout
+
+    }
+
+    private void openAddGlucoseActivity() {
+        DatabaseHandler dB = new DatabaseHandler(getApplicationContext());
+        if (dB.getUser(1) != null) {
+            // Start AddGlucose Activity passing the reading value
+            Intent intent = new Intent(getApplicationContext(), AddGlucoseActivity.class);
+            Bundle bundle = new Bundle();
+            String currentGlucose = mResult.history.get(mResult.history.size()-1).glucose(false);
+            bundle.putString("reading", currentGlucose + "");
+            intent.putExtras(bundle);
+            startActivity(intent);
+            FreestyleLibre.this.finish();
+        } else {
+            Intent intent = new Intent(getApplicationContext(), HelloActivity.class);
```

Comment:
@emartynov we can't add the reading because the user didn't create an user entry yet.
We can solve this passing the reading as extra to HelloActivity, then again to MainActivity and finally to AddGlucoseActivity. But as you can see this is a borderline case.